omxplayer/player.py

```diff
@@ -54,23 +56,31 @@ def __init__(self, filename,
                  cleaner=FileCleaner('/tmp/*omxplayer*')):
         logger.debug('Instantiating OMXPlayer')
         self.cleaner = cleaner
-        self._clean_old_files()
-
         self.args = args
-
-        if not bus_address_finder:
-            bus_address_finder = omxplayer.bus_finder.BusFinder()
-        if not Connection:
-            Connection = DBusConnection
-
         self.tries = 0
         self._is_playing = True
         self._filename = filename
+        self._Connection = Connection if Connection else DBusConnection
+        self._bus_address_finder = bus_address_finder if bus_address_finder else BusFinder()
+
+        # events
+        self.pauseEvent = Event()
+        self.playEvent = Event()
+        self.stopEvent = Event()
+        self.seekEvent = Event()
+        self.positionEvent = Event()
+
+        self._process = None
+        self._connection = None
+        self.load(filename)
+
+    def _load_file(self, filename):
+        if self._process:
+            self.quit()
+
+        self._clean_old_files()
         self._process = self._setup_omxplayer_process(filename)
-        self.connection = self._setup_dbus_connection(Connection,
-                                                     bus_address_finder)
-        time.sleep(0.5)  # Wait for the DBus interface to be initialised
-        self.pause()
+        self._connection = self._setup_dbus_connection(self._Connection, self._bus_address_finder)
```

Comment:
Note to self: Passing in a class feels a bit icky, perhaps refactor this out so that the instantiation is done outside.